### PR TITLE
Recover from metadata null key

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Fix the issue https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/2499 by allowing the WCMetaData.key field to be nullable. Since Metadata is a data that can be modified at will-be store plugins, we're seeing Gson ignoring Kotlin constraints about nullability and setting a null value within the WCMetaData model.

How to Test
==========
1. Simply ensure that the new test covering the null key scenario is working as expected, or forcefully inject a metadata with a null key inside the `StripOrderMetaData.invoke` call through the orderDto.meta_data parameter and verify it will be removed with success from the function output.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/StripOrderMetaDataTest.kt [hunk 1]
```diff
@@ -236,4 +236,48 @@ class StripOrderMetaDataTest {
             )
         )
     }
+
+    @Test
+    fun `when Metadata key is null, then remove it from the list`() {
+        // Given
+        val rawMetadata = listOf(
+            WCMetaData(
```
Comment (nbradbury) [inline plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/StripOrderMetaDataTest.kt:244]: @ThomazFB Should we include a third item here with a key that starts with an underscore?

**Comment by nbradbury** [comment]
@ThomazFB Changes are good, I just had one suggestion for the tests. Also, this is failing CI with:

```
What went wrong:
--
  | [2022-08-15T12:50:19Z] Execution failed for task ':fluxc-annotations:prepareToPublishToS3'.
  | [2022-08-15T12:50:19Z] > 'org.wordpress.fluxc.fluxc-annotations:2500-5ae27376db2057f94a4e28f3629fc2a007494253' is already published to S3!
```

**Comment by ThomazFB** [comment]
@nbradbury I've submitted your test suggestion, makes perfect sense, thanks! About the PR checks error, it seems like an issue with the artifact publishing step, I'm not sure if we can do something about it. @oguzkocer do you know anything about what's going on here? 😅

**Comment by oguzkocer** [comment]
@ThomazFB It looks like there was a network error in one of the earlier runs during the publish step and then there was a full rebuild of the current commit which I am afraid doesn't work with publishing to S3 as CI will fail if the artifact is already published.

To address the issue, you'll need to create a new commit. Probably the most useful commit in this case would be to merge the current `trunk` to this branch, but any new commit would do. Hope that helps!

**Comment by ThomazFB** [comment]
Thanks, @oguzkocer, for the context! The S3 error stopped.

@nbradbury I think we can move forward with this PR. The connected tests failing here are present in the `trunk` for more than one week, and it's not related to this PR code. WDYT?

**Comment by nbradbury** [comment]
> @nbradbury I think we can move forward with this PR.

@ThomazFB Just so I'm clear, you think it's fine to merge this PR?

**Comment by ThomazFB** [comment]
@nbradbury yes! It seems that the CI errors displayed here are coming from the trunk, so I think it doesn't make sense to fix them in this PR. Also, I already addressed the unit test adjustment you suggested.

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**